docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,397 @@
+# We create the `DofHandler` and then add a single scalar field called `:u` based on
+# our interpolation `ip` defined above.
+# Lastly we `close!` the `DofHandler`, it is now that the dofs are distributed
+# for all the elements.
+dh = DofHandler(grid)
+add!(dh, :u, ip)
+close!(dh);
+
+# Now that we have distributed all our dofs we can create our tangent matrix,
+# using `create_sparsity_pattern`. This function returns a sparse matrix
+# with the correct entries stored. We need to pass the topology and the cross-element coupling matrix when we're using
+# discontinuous interpolations. The cross-element coupling matrix is of size [1,1] in this case as
+# we have only one field and one DofHandler.
+K = create_sparsity_pattern(dh, topology = topology, cross_coupling = trues(1,1));
+
+# ### Boundary conditions
+# The Dirichlet boundary conditions are treated 
+# as usual by a `ConstraintHandler`.
+ch = ConstraintHandler(dh)
+add!(ch, Dirichlet(:u, getfaceset(grid, "right"), (x, t) -> 1.0))
+add!(ch, Dirichlet(:u, getfaceset(grid, "left"), (x, t) -> -1.0))
+close!(ch);
+
+# Furthermore, we define $\partial \Omega_n$ as the `union` of the face sets with Neumann boundary conditions for later use
+∂Ωₙ = union(
+    getfaceset(grid, "top"),
+    getfaceset(grid, "bottom"),
+);
+
+
+# ### Assembling the linear system
+#
+# Now we have all the pieces needed to assemble the linear system, $K u = f$.
+# Assembling of the global system is done by looping over all the elements in order to
+# compute the element contributions ``K_e`` and ``f_e``, all the interfaces
+# to compute their contributions ``K_i``, and all the Neumann boundary faces to compute their
+# contributions ``f_e`` which are then assembled to the
+# appropriate place in the global ``K`` and ``f``.
+#
+# #### Element assembly
+# We define the functions
+# * `assemble_element!` to compute the contributions ``K_e`` and ``f_e`` of volume integrals over an element using `cellvalues`.
+# * `assemble_interface!` to compute the contribution ``K_i`` of surface integrals over an interface using `interfacevalues`.
+# * `assemble_boundary!` to compute the contribution ``f_e`` of surface integrals over a boundary face using `facevalues`.
+#
+# The function takes pre-allocated `ke`, `ki`, and `fe` (they are allocated once and
+# then reused for all elements) so we first need to make sure that they are all zeroes at
+# the start of the function by using `fill!`. Then we loop over all the quadrature points,
+# and for each quadrature point we loop over all the (local) shape functions. The *values
+# objects provide values needed to calculate contributions as follows:
+# * `cellvalues`:
+#     * test function value ($\delta u$)
+#     * test and trial functions gradient ($\nabla u$ and $\nabla \delta u$)
+# * `facevalues`:
+#     * test function value ($δu$)
+# * `interfacevalues`:
+#     * jump in test and trial functions values
+#     * average of test and trial functions gradients
```

Comment:
I think with the additions in the bullet list above, these parts aren't required here. If desired, a few notes on `InterfaceValues` could be nice to keep, but the main info could come from referring to their docstring. 
```suggestion
```